document/core/exec/numerics.rst

```diff
@@ -1993,81 +1988,170 @@ each environment globally chooses a fixed projection for each operator.
    one particular choice by the execution environment. The fixed operation
    itself can still be non-deterministic or partial.
 
-.. todo::
-   Below is a draft of concrete definitions for relaxed-simd operations. The
-   text description is an informal description of the instructions and are not
-   the final text.
 
-Relaxed Multiply Add (madd) and Relaxed Negative Multiply Add (nmadd)
-allows for fused or unfused results. :math:`fma` is defined by |IEEE754|_
-(Section 5.4.1) as *fusedMultiplyAdd*.
+.. _op-frmadd:
+
+:math:`\frmadd_N(z_1, z_2, z_3)`
+................................
+
+Relaxed Multiply Add (madd) allows for fused or unfused results. :math:`fma` is
+defined by |IEEE754|_ (Section 5.4.1) as *fusedMultiplyAdd*.
```

Comment:
```suggestion
Relaxed multiply-add allows for fused or unfused results. The function :math:`\X{fma}` is
defined by |IEEE754|_ (Section 5.4.1) as *fusedMultiplyAdd*.
```
Despite this citation, I'm afraid we need a full definition here. The Wasm spec is ought to be self-contained.

Reply:
Hm, IEEE754 doesn't really specify this, just says "computed to unbounded precision"
@Maratyszcza any ideas how we can specify FMA here?

Reply:
https://webassembly.github.io/spec/core/exec/numerics.html#floating-point-operations
The spec defines the floating point format, but not floating point arithmetics operations, can we similarly punt fusedMultiplyAdd definition to IEEE-754?

Reply:
The spec specifies all arithmetic operations to the last bit, mainly in terms of real arithmetics and rounding. This should suffice for FMA as well, if I understand its semantics correctly. (The more involved part probably is writing suitable tests.)

Reply:
I have some tests with special values to check for FMA https://github.com/WebAssembly/relaxed-simd/blob/main/test/core/relaxed-simd/relaxed_madd_nmadd.wast